swift/Sources/FlatBuffers/ByteBuffer.swift

```diff
@@ -350,10 +350,27 @@ public struct ByteBuffer {
   /// - Parameters:
   ///   - def: Type of the object
   ///   - position: the index of the object in the buffer
+#if swift(<5.7)
   @inline(__always)
   public func read<T>(def: T.Type, position: Int) -> T {
     _storage.memory.advanced(by: position).load(as: T.self)
   }
+#else
+  @inline(__always)
+  @_specialize(exported: true, where T==Int8)
```

Comment:
nit: I think this is also missing a `bool` (not sure if we read bools directly)

Reply:
I am afraid, we also use `read` for `NativeStruct`, but I am not 100% sure about it.